java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -0,0 +1,330 @@
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.widget.ArrayAdapter;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.support.design.widget.BottomNavigationView;
+
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.HealthVaultInitializationHandler;
+import com.microsoft.hsg.android.simplexml.HealthVaultSettings;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+import com.microsoft.hsg.android.simplexml.client.RequestCallback;
+import com.microsoft.hsg.android.simplexml.things.types.types.PersonInfo;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+
+public class MainActivity 
+	extends ListActivity 
+	implements HealthVaultInitializationHandler {
+
+	private HealthVaultApp service;
+	private HealthVaultClient hvClient;
+	private BottomNavigationView mBottomNav;
+	private LinearLayout weightTile;
+
+	private ProgressDialog connectProgressDialog;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_main);
+		
+		service = HealthVaultApp.getInstance(this);
+		if (service.isAppConnected()) {
+			service.start(this, this);
+		}
+
+		hvClient = new HealthVaultClient();
+		
+		mBottomNav = (BottomNavigationView) findViewById(R.id.navigation);
+
+		mBottomNav.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
+			@Override
+			public boolean onNavigationItemSelected(MenuItem item) {
+				// handle desired action here
+				// One possibility of action is to replace the contents above the nav bar
+				// return true if you want the item to be displayed as the selected item
+				return true;
+			}
+		});
+
+		weightTile = (LinearLayout) findViewById(R.id.WeightTile);
+		weightTile.setOnClickListener(new View.OnClickListener() {
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (service.isAppConnected()) {
+					intent = new Intent(MainActivity.this, WeightActivity.class);
+				}
+				else
+				{
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
+					startActivity(intent);
+				}
+			}
+		});
+	}
+	
+	@Override
+	protected void onStart() {
+		super.onStart();
+		hvClient.start();
+	}
+	
+	@Override
+    protected void onResume()
+    {
+		super.onResume();
+    }
+	
+    @Override
+	protected void onStop() {
+		super.onStop();
+		if(connectProgressDialog != null) {
+			connectProgressDialog.dismiss();
+		}
+		
+		hvClient.stop();
+	}
+    
+    @Override
```

Comment:
This seems to be indented correctly

---
In reply to: [112363525](https://github.com/Microsoft/healthvault-samples/pull/4#discussion_r112363525) [](ancestors = 112363525)